### PR TITLE
Azure Relay & it's entities

### PR DESCRIPTION
**Pull request by grandhiramesh**
### Contributing guide

https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md
### Changelog

*
*
*
*
*
### Description of the change

Azure Relay & it's entities
- Create Namespace
- Create HybridConnection
- Create WCFRelay
- Create SAS Policies

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@grandhiramesh__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

**Comment by grandhiramesh** [comment]
Azure Relay Service is going live (public preview) early November. Can anyone please take a look at the earliest

**Comment by azurecla** [comment]
Hi __@grandhiramesh__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (rgrandhi). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by grandhiramesh** [comment]
Updated PR based on the review comments

File: 201-azurerelay-create-all-resources/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+            "name": "[parameters('wcfRelayName')]",
+            "type": "wcfRelays",
+            "dependsOn": [
+                "[concat('Microsoft.Relay/namespaces/', parameters('namespaceName'))]"
+            ],
+            "properties": {
+                "path": "[parameters('wcfRelayName')]",
+                "relayType": "[parameters('wcfRelayType')]"
+            }
+        }, {
```
Comment (gatneil) [inline 201-azurerelay-create-all-resources/azuredeploy.json]: formatting still seems a bit off :)

File: 101-azurerelay-create-namespace/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        }
+    },
+    "variables": {
+        "location": "[resourceGroup().location]",
+        "apiVersion": "2016-07-01",
+        "defaultSASKeyName": "RootManageSharedAccessKey",
+        "defaultAuthRuleResourceId": "[resourceId('Microsoft.Relay/namespaces/authorizationRules', parameters('name'), variables('defaultSASKeyName'))]"
+    },
+    "resources": [{
+        "apiVersion": "[variables('apiVersion')]",
```
Comment (gatneil) [inline 101-azurerelay-create-namespace/azuredeploy.json]: please use a raw string instead of a variable (here and in all other apiVersions)

**Comment by grandhiramesh** [comment]
ApiVersion is intended to call the correct API based on the version provided by the enduser.

We intend to keep it as a variable as it would help us to extend it in later point of time when a new version is released.  In later releases, it’s easier for us to control by introducing parameter and assign the parameter value to the variable. Helps us in backward compatibility too.

**Comment by singhkays** [comment]
@grandhiramesh In the template folder naming, please use "azure-relay" instead of "azurerelay"

**Comment by grandhiramesh** [comment]
Renamed the folder from "azurerelay" to "azure-relay"

**Comment by singhkays** [comment]
lgtm, merged

**Pull request merged**